astropy/table/tests/test_table.py

```diff
@@ -1364,3 +1372,92 @@ def test_nested_iteration():
         for r2 in t:
             out.append((r1['a'], r2['a']))
     assert out == [(0, 0), (0, 1), (1, 0), (1, 1)]
+
+
+class TestPandas(object):
+
+    @pytest.mark.skipif('not HAS_PANDAS')
+    def test_simple(self):
+
+        from pandas import DataFrame
+
+        t = table.Table()
+
+        for endian in ['<', '>']:
+            for kind in ['f', 'i']:
+                for byte in ['2','4','8']:
+                    dtype = np.dtype(endian + kind + byte)
+                    x = np.array([1,2,3], dtype=dtype)
+                    t[endian + kind + byte] = x
+
+        t['u'] = ['a','b','c']
+        t['s'] = [b'a', b'b', b'c']
+
+        d = t.to_pandas()
+
+        for column in t.columns:
+            if column == 'u':
+                assert np.all(t['u'] == np.array(['a','b','c']))
+            elif column == 's':
+                assert np.all(t['s'] == np.array([b'a',b'b',b'c']))
+            else:
+                assert_allclose(t[column], d[column])
```

Comment:
@taldcroft - based on the above, if I add columns one by one to the data frame then for byte arrays it will actually have the correct type. For unicode I think it's always going to be ``O``